book/src/build_system.md

```diff
@@ -51,6 +52,79 @@ Modules available in Ariel OS include:
 ## laze contexts
 
 The laze configuration defines a laze context for each MCU, MCU family, and board.
+These can be found in the [support matrix](./hardware_functionality_support.html), where they are called “Ariel OS name”.
 
-laze passes the names of all contexts related to the selected builder as rustc `--cfg context=$CONTEXT` flags.
+Out-of-tree applications can be restricted to specific laze contexts, see [below](#restricting-an-application-to-specific-mcusboards).
+
+In addition, laze passes the names of all contexts related to the selected builder as rustc `--cfg context=$CONTEXT` flags.
 This makes it possible to use the `#[cfg]` attribute to introduce feature-gates based on the MCU, MCU family, or board, when required.
+
+## Out-of-tree applications
+
+Out-of-tree applications use the `laze-project.yml` file for configuration through laze.
+
+### Importing Ariel OS
+
+Ariel OS's source and configuration are imported using [laze's `imports`][laze-imports-book] feature.
+The [project templates](./getting_started.md#starting-an-application-project-from-a-template-repository) use a [`git` import][laze-git-import-book] to ask laze to clone Ariel OS's repository.
+The cloned repository is stored inside `build/imports`.
+
+> It is currently recommended to use Ariel OS's commit ID to track the repository, to avoid surprising changes.
+> This commit ID needs to be updated to update the version of Ariel OS used by the application.
+
+It is alternatively possible to clone the repository manually and specify the resulting directory using a [`path` import][laze-path-import-book].
+This can be useful when needing to modify Ariel OS itself, when also working on an application.
+
+### Enabling laze modules for an application
+
+Instead of manually specifying [laze modules on the command line](#laze-modules), laze modules required for an application must be specified in the application's laze configuration file, `laze-project.yml`.
```

Comment:
Bit nitpicky here, I would swap the ordering here and *not* start with what must not be done. 
```suggestion
Laze modules required for an application must be specified in the application's laze configuration file, `laze-project.yml`, instead of manually specifying [laze modules on the command line](#laze-modules).
```